foyer/general_forcefield.py

```diff
@@ -0,0 +1,539 @@
+        """
+        if self.atomTypeDefinitions == {}:
+            raise FoyerError('Attempting to atom-type using a forcefield '
+                'with no atom type definitions.')
+
+        if backend == 'gmso':
+            if not isinstance(top, gmso.Topology):
+                mb = import_('mbuild')
+                tmp_top = mb.load(top)
+                top = gmso.external.from_mbuild(top)
+
+            assert isinstance(top, gmso.Topology)
+            typemap = self._run_atomtyping(top,
+                    use_residue_map=use_residue_map,
+                    **kwargs)
+
+        return self._parametrize(top=top, typemap=typemap,
+            references_file=references_file,
+            assert_bond_params=assert_bond_params,
+            assert_angle_params=assert_angle_params,
+            assert_dihedral_params=assert_dihedral_params,
+            assert_improper_params=assert_improper_params,
+            verbose=verbose,
+            backend=backend,
+            combining_rule=combining_rule,
+            *args, **kwargs)
+
+    def _run_atomtyping(self, top, use_residue_map=True, **kwargs):
+        """Atomtype the topology
+
+        Parameters
+        ----------
+        top : gmso.Topology
+            Molecular Topology to find atom types for.
+        use_residue_map : bool, optional, default=True
+            Spped up options for duplicates subtopology.
+        """
+
+        if isinstance(top, mb.Compound):
+            top = from_mbuild(top)
+        # TO DO in another PR
+        if use_residue_map:
+            # Detect duplicates subtopology/residues
+            # (do matching by name, assert same number
+            # of atoms)
+            # Not implemented yet
+            typemap = find_atomtypes(top, forcefield=self)
+        else:
+            typemap = find_atomtypes(top, forcefield=self)
+
+        return typemap
+
+    def _parametrize(self, top=None,
+                        typemap=dict(),
+                        references_file=None,
+                        assert_bond_params=True,
+                        assert_angle_params=True,
+                        assert_dihedral_params=True,
+                        assert_improper_params=True,
+                        verbose=False,
+                        backend='gmso',
+                        combining_rule='geometric',
+                        debug=False,
+                        **kwargs):
+        """Parametrize the Topology from the typemap provided
+
+        Assign AtomTypes and BondTypes to Atoms and Bonds, respectively.
+        Creat Angles, Dihedrals, Impropers and assing corresponding
+        AngleType, DihedralTypes, and ImproperTypes.
+
+        Parameters
+        ----------
+        top : gmso.Topology
+            gmso.Topology object that needed to be parametrized
+        typemap : dict
+            typemap generated by the atomtyper
+        references_file : str, optional, defaut=None
```

Comment:
Spelling/grammer suggestion: "Name of file where force field refrences will be written", where "refrences" is misspelled.

Suggested: "Name of the file where force field references will be written"